internal/sdkprovider/service/kafkatopic/kafka_topic.go

```diff
@@ -265,22 +265,15 @@ func resourceKafkaTopicCreate(ctx context.Context, d *schema.ResourceData, m int
 	topicName := d.Get("topic_name").(string)
 	partitions := d.Get("partitions").(int)
 	replication := d.Get("replication").(int)
+	client := m.(*aiven.Client)
 
 	// aiven.KafkaTopics.Create() function may return 501 on create
 	// Second call might say that topic already exists
 	// So to be sure, better check it before create
-	_, err := getTopic(ctx, m, d.Timeout(schema.TimeoutRead), project, serviceName, topicName)
-
-	// No error means topic exists
-	if err == nil {
+	if isTopicExists(ctx, client, project, serviceName, topicName) {
```

Comment:
nit: consider adding a comment above this block to briefly explain its purpose for clarity

Reply:
We have a big comment here already 
```
// aiven.KafkaTopics.Create() function may return 501 on create
	// Second call might say that topic already exists
	// So to be sure, better check it before create
```